src/test/java/org/jabref/logic/importer/fileformat/GrobidPdfMetadataImporterTest.java

```diff
@@ -0,0 +1,71 @@
+package org.jabref.logic.importer.fileformat;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.logic.importer.ImportFormatPreferences;
+import org.jabref.logic.importer.fetcher.GrobidCitationFetcher;
+import org.jabref.logic.util.StandardFileType;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Answers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class GrobidPdfMetadataImporterTest {
+
+    private GrobidPdfMetadataImporter importer;
+    private ImportFormatPreferences importFormatPreferences;
+
+    @BeforeEach
+    public void setUp() {
+        importFormatPreferences = mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS);
+        when(importFormatPreferences.getKeywordSeparator()).thenReturn(',');
+        importer = new GrobidPdfMetadataImporter(GrobidCitationFetcher.GROBID_URL, importFormatPreferences);
+    }
+
+    @Test
+    public void testsGetExtensions() {
+        assertEquals(StandardFileType.PDF, importer.getFileType());
+    }
+
+    @Test
+    public void testImportEntries() throws URISyntaxException {
+        Path file = Path.of(GrobidPdfMetadataImporterTest.class.getResource("LNCS-minimal.pdf").toURI());
+        List<BibEntry> bibEntries = importer.importDatabase(file, StandardCharsets.UTF_8).getDatabase().getEntries();
+
+        assertEquals(1, bibEntries.size());
+
+        BibEntry be0 = bibEntries.get(0);
+        assertEquals(Optional.of("Lastname, Firstname"), be0.getField(StandardField.AUTHOR));
+        assertEquals(Optional.of("Paper Title"), be0.getField(StandardField.TITLE));
```

Comment:
If I understand grobid correctly, it is basically a machine-learning search algorithm. I think the more general the search query, the more vague the result may be, as long as grobid is not yet so well trained. On the other hand, the more specific the search query, the more specific the result should be - or am I wrong?

Reply:
So in this case the 'search query' would be the pdf. It's quite minimalistic so I expect Grobid to detect this information. I think the tests are ok, but maybe I should add a note (as a comment) so that if it breaks in the future, people know what to do.